### PR TITLE
Convert the rest of tests to mocha

### PR DESCRIPTION
**Pull request by ezubarev**
The remaining tests were ported, nodeunit was completely removed. Have a look. Resolves #849
 @megawac, @aearly

### CODE REVIEW

File: mocha_test/timeout.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 var async = require('../lib');
-var expect = require('chai').expect;
+var assert = require('assert');
```
Comment (megawac) [inline mocha_test/timeout.js]: why this change? Won't this cause errors to be less easy to debug?

Comment (ezubarev) [inline mocha_test/timeout.js]: Sure, you're right. I'll fix it.

File: mocha_test/asyncify.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+var async = require('../lib');
+var assert = require('assert');
```
Comment (megawac) [inline mocha_test/asyncify.js]: same here, use `chai.expect` instead of assert please

**Comment by ezubarev** [comment]
Assert in wrapped function could not be handled which led the test result to false positive. Fixed in 210de70.

**Comment by ezubarev** [comment]
@megawac, I replaced assert with expect in tests on asyncify. In other cases as I can see it's usage will not lead to less transparency in debugging. However I don't mind if you want to make use `chai.expect` somewhere else.

**Comment by megawac** [comment]
I'm sure any confusing remaining ones will be fixes post merge. Thanks a bunch @ezubarev

**Comment by ezubarev** [comment]
You're welcome.
